src/lib/Key.js

```diff
@@ -11,16 +17,49 @@ class Key {
         return Nimiq.BufferUtils.toBase64(Nimiq.Hash.computeBlake2b(input));
     }
 
+    /**
+     * @returns {EncryptionKeyParams}
+     */
+    static get defaultEncryptionKeyParams() {
+        return {
+            kdf: CONFIG.RSA_KDF_FUNCTION,
+            iterations: CONFIG.RSA_KDF_ITERATIONS,
+            keySize: CONFIG.RSA_KEY_BITS,
+        };
+    }
+
+    /**
+     * @param {EncryptionKeyParams} paramsA
+     * @param {EncryptionKeyParams} paramsB
+     * @returns {boolean}
+     */
+    static _isEncryptionKeyParamsEqual(paramsA, paramsB) {
+        return typeof paramsA === 'object' && typeof paramsB === 'object'
+            && paramsA.kdf === paramsB.kdf
+            && paramsA.iterations === paramsB.iterations
+            && paramsA.keySize === paramsB.keySize;
+    }
+
+    /**
+     * @param {EncryptionKeyParams} params
+     * @returns {boolean}
+     */
+    static _isDefaultEncryptionKeyParams(params) {
+        return Key._isEncryptionKeyParamsEqual(params, Key.defaultEncryptionKeyParams);
+    }
+
     /**
      * @param {Nimiq.Entropy|Nimiq.PrivateKey} secret
-     * @param {boolean} [hasPin]
+     * @param {KeyConfig} [config]
      */
-    constructor(secret, hasPin = false) {
+    constructor(secret, config = {}) {
         this._secret = secret;
-        this._hasPin = hasPin;
+        this._hasPin = Boolean(config.hasPin);
         /** @type {string?} */
         this._id = null;
         this._defaultAddress = this.deriveAddress(Constants.DEFAULT_DERIVATION_PATH);
+
+        this.rsaKeyPair = config.rsaKeyPair;
```

Comment:
```suggestion
        this._rsaKeyPair = config.rsaKeyPair;
```
For consistency with other properties, the name should start with an underscore, and then a getter be added. Also, I would move the definition just below the definition of `_hasPin`, as it also belongs to the `config`.

Additonal getter:
```js
    /**
     * @returns {RsaKeyPairExport | undefined}
     */
    getRsaKeyPairExportIfExists() {
        return this._rsaKeyPair;
    }
```

Reply:
Resolved in #524